jdaviz/configs/cubeviz/plugins/spectral_extraction/spectral_extraction.py

```diff
@@ -214,26 +245,56 @@ def collapse_to_spectrum(self, add_data=True, **kwargs):
             nddata = spectral_cube.get_subset_object(
                 subset_id=self.aperture.selected, cls=NDDataArray
             )
-            uncertainties = uncert_cube.get_subset_object(
-                subset_id=self.aperture.selected, cls=StdDevUncertainty
+            if uncert_cube:
+                uncertainties = uncert_cube.get_subset_object(
+                    subset_id=self.aperture.selected, cls=StdDevUncertainty
+                )
+            # Exact slice mask of cone or cylindrical aperture through the cube. `shape_mask` is
+            # a 3D array with fractions of each pixel within an aperture at each
+            # wavelength, on the range [0, 1].
+            if self.function_selected.lower() in ['min', 'max'] and self.wavelength_dependent:
+                shape_mask = self.cone_aperture().astype(int)
+            elif self.function_selected.lower() in ['min', 'max']:
+                shape_mask = self.cylindrical_aperture().astype(int)
+            elif self.wavelength_dependent:
+                shape_mask = self.cone_aperture()
+            else:
+                shape_mask = self.cylindrical_aperture()
+
+            if self.aperture_method_selected.lower() == 'center':
+                flux = nddata.data << nddata.unit
+            else:
+                # Apply the fractional pixel array to the flux cube
+                flux = (shape_mask * nddata.data) << nddata.unit
+            # Boolean cube which is True outside of the aperture
+            # (i.e., the numpy boolean mask convention)
+            mask = np.isclose(shape_mask, 0)
+
+        elif self.aperture.selected != self.aperture.default_text:
```

Comment:
should/can this block be covered in tests? 😬

Reply:
ah, tests are follow-up, as long as @pllim can make sure this gets covered in the science validation tests, then this can be deferred for this PR

Reply:
What is `self.aperture.selected != self.aperture.default_text` supposed to check for?

Reply:
Wait, isn't this check a duplicate of L244 ? Should this whole block be deleted?

Reply:
Good catch.